Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/FindPeakAutomatic.py

```diff
@@ -0,0 +1,395 @@
+        self._bad_peak_to_consider = self.getProperty('BadPeaksToConsider').value
+        self._use_poisson_cost = self.getProperty('UsePoissonCost').value
+        self._fit_to_baseline = self.getProperty('FitToBaseline').value
+        self._plot_peaks = self.getProperty('PlotPeaks').value
+        self._plot_baseline = self.getProperty('PlotBaseline').value
+        self._estimate_peak_sigma = self.getProperty('EstimatePeakSigma').value
+        self._min_sigma = self.getProperty('MinPeakSigma').value
+        self._max_sigma = self.getProperty('MaxPeakSigma').value
+
+        if self._max_sigma < self._min_sigma:
+            issues['MinPeakSigma'] = 'Sigma bounds must be: MinPeakSigma <= MaxPeakSigma'
+            issues['MaxPeakSigma'] = 'Sigma bounds must be: MinPeakSigma <= MaxPeakSigma'
+
+        if self._estimate_peak_sigma < self._min_sigma:
+            issues['EstimatePeakSigma'] = 'EstimatePeakSigma must be greater than MinPeakSigma'
+
+        if self._estimate_peak_sigma > self._max_sigma:
+            issues['EstimatePeakSigma'] = 'EstimatePeakSigma must be greater than MaxPeakSigma'
+
+        return issues
+
+    def PyExec(self):
+        # Progress reporter for algorithm initialization
+        prog_reporter = Progress(self, start=0.0, end=0.1, nreports=4)
+
+        raw_xvals, raw_yvals, raw_error, error_ws = self.load_data(prog_reporter)
+
+        # Convert the data to point data
+        prog_reporter.report('Converting to point data')
+        raw_data_ws = ConvertToPointData(error_ws, StoreInADS=False)
+        raw_xvals = raw_data_ws.readX(0).copy()
+        raw_yvals = raw_data_ws.readY(0).copy()
+
+        raw_xvals, raw_yvals, raw_error = self.crop_data(raw_xvals, raw_yvals, raw_error, prog_reporter)
+
+        # Find the best peaks
+        (peakids, peak_table,
+         refit_peak_table), baseline = self.process(raw_xvals,
+                                                    raw_yvals,
+                                                    raw_error,
+                                                    acceptance=self._acceptance,
+                                                    average_window=self._smooth_window,
+                                                    bad_peak_to_consider=self._bad_peak_to_consider,
+                                                    use_poisson=self._use_poisson_cost,
+                                                    peak_width_estimate=self._estimate_peak_sigma,
+                                                    fit_to_baseline=self._fit_to_baseline,
+                                                    prog_reporter=prog_reporter)
+
+        if self._plot_peaks:
+            self.plot_peaks(raw_xvals, raw_yvals, baseline, peakids)
+
+        self.set_output_properties(peak_table, refit_peak_table)
+
+    def load_data(self, prog_reporter):
+        # Load the data and clean from Nans
+        raw_xvals = self.getProperty('InputWorkspace').value.readX(0).copy()
```

Comment:
I think the idea here is to create a new workspace if the original does not have errors? In this case we are creating a workspace using `CreateWorkspace` that the property system has no idea about so this algorithm cannot be used as child algorithm without having to clear up after a possible new workspace that would pop into the ADS.

Workspaces that are created in algorithms should go through the property mechanism so that the outer algorithm can know whether they are to be stored in the ADS or not. Can I suggest that we add an `OutputWorkspace` property and set that either to the input workspace, if it has errors, or the new workspace with errors added if one is created?